pkg/agent/agent.go

```diff
@@ -123,7 +123,16 @@ func (k *Klocksmith) process(stop <-chan struct{}) error {
 			return err
 		}
 	} else {
-		glog.Info("Skipping marking node as schedulable -- node was unschedulable prior to update reboot")
+		glog.Info("Skipping marking node as schedulable -- node was marked unschedulable by an external source")
+	}
+
+	anno = map[string]string{
+		constants.AnnotationAgentMadeUnschedulable: constants.False,
+	}
+
+	glog.Infof("Setting annotations %#v", anno)
+	if err := k8sutil.SetNodeAnnotations(k.nc, k.node, anno); err != nil {
```

Comment:
We only need to do this if we're the ones who marked the node unschedulable, right? I.e. we can move this hunk to under the `if makeSchedulable {` above.

Reply:
It seems like it can be moved. I agree with this.